File: src/main/java/facts/stmt/Stmt.java
package facts.stmt;

import facts.Fact;
import facts.Label;
import java.util.List;

/**
 * Generic instruction fact
 */
public abstract class Stmt extends Fact {

    private Label label;

    public Stmt(String pc) {
        label = new Label(pc);
    }

    public Label getLabel() {
        return label;
    }

    public void setLabel(Label label) {
        this.label = label;
    }

    public abstract List<Fact> toFactList();
}

File: src/main/scala/astnodes/stmt/Stmt.scala
package astnodes.stmt

import astnodes.exp.Expr
import astnodes.exp.variable.Variable

/** Generic instruction fact
  */
trait Stmt(val pc: String, val labelVisible: Boolean = false) {
  def subst(v: Variable, w: Variable): Stmt

  def toBoogieString: String = toString

  def labelString: String = if (labelVisible) {
    "label" + pc + ": "
  } else {
    ""
  }

  //def withVisibleLabel: Stmt
}